### PR TITLE
Fleshing out S3 storage CompleteConfiguration function

### PR DESCRIPTION
**Pull request by coreydaley**
https://jira.coreos.com/browse/DEVEXP-145

### CODE REVIEW

**Comment by coreydaley** [comment]
@legionus @dmage ptal
I know this isn't 100% done yet, but I would appreciate your opinion on what I have done thus far.

File: pkg/storage/s3/s3.go [hunk 1]
```diff
@@ -59,7 +68,75 @@ func (d *driver) Volumes() ([]corev1.Volume, []corev1.VolumeMount, error) {
...
+	return err
+}
+
+// createBucket attempts to create an s3 bucket with the given name
+func (d *driver) createBucket(svc *s3.S3) error {
+	input := &s3.CreateBucketInput{
+		Bucket: aws.String(d.Config.Bucket),
+	}
+
+	if len(d.Config.Region) > 0 && d.Config.Region != "us-east-1" {
```
Comment (dmage) [inline pkg/storage/s3/s3.go]: Why should the "us-east-1" region be a special case?

File: pkg/apis/imageregistry/v1alpha1/types.go [hunk 2]
```diff
@@ -36,6 +36,8 @@ type ImageRegistryConfigStorageS3 struct {
 	Region         string `json:"region,omitempty"`
 	RegionEndpoint string `json:"regionEndpoint,omitempty"`
 	Encrypt        bool   `json:"encrypt,omitempty"`
+	AccessKey      string `json:"accesskey,omitempty"`
+	SecretKey      string `json:"secretkey,omitempty"`
```
Comment (dmage) [inline pkg/apis/imageregistry/v1alpha1/types.go]: @legionus @bparees are we ok with having secrets in our custom resource?

File: pkg/storage/s3/s3.go (hunk 1 above)
Comment (coreydaley) [reply pkg/storage/s3/s3.go]: It's a weird AWS API thing where us-east-1 is a special case.

File: pkg/apis/imageregistry/v1alpha1/types.go (hunk 2 above)
Comment (bparees) [reply pkg/apis/imageregistry/v1alpha1/types.go]: no, they need to be references to a secret.

**Comment by bparees** [comment]
@legionus any comments?

**Comment by bparees** [comment]
(lgtm, but since it's integrating with the pattern @legionus established, he should sign off)

**Comment by coreydaley** [comment]
@bparees @legionus @dmage I will clean up the test errors tonight.

**Comment by coreydaley** [comment]
@legionus @dmage @bparees ptal
I have updated the code and fixed the tests

File: test/e2e/s3_test.go [hunk 3]
```diff
@@ -0,0 +1,25 @@
...
+
+package e2e
+
+import (
+	"testing"
+
+	f "github.com/operator-framework/operator-sdk/pkg/test"
+)
+
+func TestS3(m *testing.M) {
```
[the remark this answers is not in the document]
Comment (coreydaley) [reply test/e2e/s3_test.go]: Good catch, updated in current push.

**Pull request merged**